### PR TITLE
Action cable

### PR DESCRIPTION
**Pull request by MarcosFierro**

### CODE REVIEW

File: app/assets/javascripts/channels/notifications_channel.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  speak: function(data) {
+    return this.perform('speak', {
+      message: data.message,
+      current_user: data.current_user
+    });
+  },
+
+  renderNotification: function(data) {
+    alert("New notification\n" + data.title + "\n" + data.description);
+    //return "<h5> <b>" + data.title + "</b> </h5> <br> <h7>" + data.descrition + "</h7>";
```
Comment (AlvaroSenorale) [inline app/assets/javascripts/channels/notifications_channel.js]: remove this code

File: app/channels/notifications_channel.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class NotificationsChannel < ApplicationCable::Channel
+  def subscribed(current_user_id = nil)
+    stream_from current_user_id ? "notifications_#{current_user_id}" : 'notifications'
```
Comment (AlvaroSenorale) [inline app/channels/notifications_channel.rb]: what does this `stream_from` ? I'm wondering why we would wanto a stream without a user_id

File: app/channels/notifications_channel.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  def subscribed
+    stream_from specific_channel
+  end
+
+  def speak(data); end
+
+  private
+
+  def specific_channel
+    "notifications_#{params[:user_id]}"
```
Comment (AlvaroSenorale) [inline app/channels/notifications_channel.rb]: shouldn't we use the cookie instead of params?

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -16,8 +15,13 @@ def create
...
 
   def destroy
-    session[:user_id] = nil
-    flash[:success] = 'You have logged out'
+    setup
     redirect_to root_path
   end
+
+  def setup(user = nil)
+    session[:user_id] = user ? user.id : nil
```
Comment (AlvaroSenorale) [inline app/controllers/sessions_controller.rb]:
```suggestion
    session[:user_id] = user&.id
```

File: app/channels/notifications_channel.rb (hunk 3 above)
Comment (MarcosFierro) [reply app/channels/notifications_channel.rb]: I think i had same problem as with session. 'cookies' is only available for controllers and/or views

File: app/channels/application_cable/connection.rb [hunk 5]
```diff
@@ -2,5 +2,20 @@
...
+    identified_by :current_user
+
+    def connect
+      self.current_user = find_verified_user
+    end
+
+    protected
+
+    def find_verified_user
+      if (current_user = User.find(cookies[:user_id]))
```
Comment (AlvaroSenorale) [inline app/channels/application_cable/connection.rb]: ```User.find(cookies[:user_id])``` fails if the user doesn't exists. Look for another AR method to do this

File: app/channels/notifications_channel.rb (hunk 3 above)
Comment (AlvaroSenorale) [reply app/channels/notifications_channel.rb]: From one of the responses here https://github.com/rails/rails/issues/30917 . Seems that you have access to `current_user` here, right?

File: config/routes.rb [hunk 6]
```diff
@@ -5,6 +5,9 @@
 Rails.application.routes.draw do
   root 'pages#index'
 
+  get 'users/:user_id/comments', to: 'api/v1/comments#comments_by_user'
+  get 'users/:user_id/notifications', to: 'api/v1/notifications#notifications_by_user', as: 'user_notifications'
```
Comment (AlvaroSenorale) [inline config/routes.rb]: ok, I think it's time to decouple the api from the other routes

File: app/channels/notifications_channel.rb (hunk 3 above)
Comment (MarcosFierro) [reply app/channels/notifications_channel.rb]: Definitely yes.

**Pull request merged**